account_invoice_alternate_payer/models/account_invoice.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = 'account.invoice'
+
+    alternate_payer_id = fields.Many2one(
+        'res.partner', string='Alternate Payer',
+        help="If set, this partner will be the that we expect to pay or to "
+             "be paid by. If not set, the payor is by default the "
+             "commercial")
+
+    @api.multi
+    def finalize_invoice_move_lines(self, move_lines):
```

Comment:
That is probably a bad choice. Now, the system is no longer usable with reimbursables. It is not probable to have a reimburasable invoice with an alternate payor, but it is good to know.

It think it could be better to use: https://github.com/odoo/odoo/blob/11.0/addons/account/models/account_invoice.py#L1189
Here, the partner is assigned to the move line. Also, you should not change the partner of all the lines, but only the line used for payment (type='dest'). Isn't it?